src/structures/APIMessage.js

```diff
@@ -0,0 +1,321 @@
+  }
+
+  /**
+   * Partitions embeds and attachments.
+   * @param {Array<MessageEmbed|MessageAttachment>} items Items to partition
+   * @returns {Array<MessageEmbed[], MessageAttachment[]>}
+   */
+  static partitionMessageAdditions(items) {
+    const embeds = [];
+    const files = [];
+    for (const item of items) {
+      if (item instanceof MessageEmbed) {
+        embeds.push(item);
+      } else if (item instanceof MessageAttachment) {
+        files.push(item);
+      }
+    }
+
+    return [embeds, files];
+  }
+
+  /**
+   * Transforms the user-level arguments into a final options object. Passing a transformed options object alone into
+   * this method will keep it the same, allowing for the reuse of the final options object.
+   * @param {StringResolvable} [content=''] Content to send
+   * @param {MessageOptions|WebhookMessageOptions|MessageAdditions} [options={}] Options to use
+   * @param {MessageOptions|WebhookMessageOptions} [extra={}] Extra options to add onto transformed options
+   * @param {boolean} [isWebhook=false] Whether or not to use WebhookMessageOptions as the result
+   * @returns {MessageOptions|WebhookMessageOptions}
+   */
+  static transformOptions(content, options, extra = {}, isWebhook = false) {
+    if (!options && typeof content === 'object' && !(content instanceof Array)) {
+      options = content;
+      content = '';
+    }
+
+    if (!options) {
+      options = {};
+    }
+
+    if (options instanceof MessageEmbed) {
+      return Object.assign(isWebhook ? { content, embeds: [options] } : { content, embed: options }, extra);
+    }
+
+    if (options instanceof MessageAttachment) {
+      return Object.assign({ content, files: [options] }, extra);
+    }
+
+    if (options instanceof Array) {
+      const [embeds, files] = this.partitionMessageAdditions(options);
+      return Object.assign(isWebhook ? { content, embeds, files } : { content, embed: embeds[0], files }, extra);
+    } else if (content instanceof Array) {
+      const [embeds, files] = this.partitionMessageAdditions(content);
+      if (embeds.length || files.length) {
+        return Object.assign(isWebhook ? { embeds, files } : { embed: embeds[0], files }, extra);
+      }
+    }
+
+    return Object.assign({ content }, options, extra);
+  }
+
+  /**
+   * Creates an `APIMessage` from user-level arguments.
+   * @param {MessageTarget} target Target to send to
+   * @param {StringResolvable} [content=''] Content to send
+   * @param {MessageOptions|WebhookMessageOptions|MessageAdditions} [options={}] Options to use
+   * @param {MessageOptions|WebhookMessageOptions} [extra={}] - Extra options to add onto transformed options
+   * @returns {MessageOptions|WebhookMessageOptions}
+   */
+  static create(target, content, options, extra = {}) {
+    const Webhook = require('./Webhook');
+    const WebhookClient = require('../client/WebhookClient');
+
+    const isWebhook = target instanceof Webhook || target instanceof WebhookClient;
```

Comment:
This can be removed to reduce duplication with `APIMessage#isWebhook` getter and checked in `transformOptions`.

Reply:
They're both static.